### PR TITLE
PR 

### PR DESCRIPTION
**Pull request by pointbazaar**

### CODE REVIEW

File: parser/src/main/parsing/statements/Stmt.c [hunk 1]
```diff
@@ -30,16 +31,27 @@ struct Stmt* makeStmt(struct TokenList* tokens, bool debug) {
...
+	if (first->kind == BREAK){
+		res->m6 = makeBreakStmt(copy, debug);
+	
+		if(res->m6 == NULL){
+			printf("expected break stmt, but was:\n");
+			list_print(copy);
+			freeTokenListShallow(copy);
+			free(res);
+			exit(1);
+		}
```
Comment (Milo-D) [inline parser/src/main/parsing/statements/Stmt.c:49]: At line 124: res->m1 not free'd.

File: transpiler/main/c_code_gen.c [hunk 2]
```diff
@@ -230,6 +231,9 @@ void transpileStmt(struct Stmt* s, struct Ctx* ctx){
 		transpileAssignStmt(s->m5, ctx);
 		fprintf(ctx->file, ";");
 		
+	}else if(s->m6 != NULL){
+		transpileBreakStmt(s->m6, ctx);
+		
 	}else{
 		printf("Error in transpileStmt\n");
```
Comment (Milo-D) [inline transpiler/main/c_code_gen.c:238]: ctx and stmt not free'd

**Review by Milo-D — CHANGES_REQUESTED** [review]
No major issues found.

File: ast/ast_reader.c [hunk 3]
```diff
@@ -631,6 +633,18 @@ struct LoopStmt* readLoopStmt(FILE* file, bool debug){
...
 	return v;
 }
+struct BreakStmt* readBreakStmt(FILE* file, bool debug){
+	
+	if(debug){ printf("readBreakStmt(...)\n"); }
+	struct BreakStmt* v = smalloc(sizeof(struct BreakStmt));
+	
+	if(fscanf(file, "BreakStmt\t") == EOF){
+		printf("Error reading BreakStmt\n");
+		exit(1);
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply ast/ast_reader.c:654]: done

File: lexer/src/test.c [hunk 4]
```diff
@@ -991,3 +994,21 @@ bool test_if_else(){
...
+	struct Token** tokens = lex(str,".test.tokens");
+	
+	bool res = true;
+
+	res &= assert(tokens[0]->kind==BREAK);
+
+	res &= assert(tokens[1]->kind==SEMICOLON);
+
+	res &= assert(tokens[2]->kind==TYPEIDENTIFIER);
+
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply lexer/src/test.c:1126]: fixed in d2491766c62e3bcd0e5725e5e2565ff318b51e2f

File: parser/src/main/parsing/statements/BreakStmt.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	}
+	
+	if(list_size(tokens) < 3){ return NULL; }
+
+	struct BreakStmt* res = smalloc(sizeof(struct BreakStmt));
+
+	struct TokenList* copy = list_copy(tokens);
+
+	if(!list_expect(copy, BREAK)){
+		free(res);
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply parser/src/main/parsing/statements/BreakStmt.c]: fixed in aa3c7f449772e142e02918070c49242a18120f6e

File: parser/src/main/parsing/statements/BreakStmt.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+		return NULL;
+	}
+
+	if(!list_expect(copy, SEMICOLON)){
+		free(res);
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply parser/src/main/parsing/statements/BreakStmt.c]: fixed in aa3c7f449772e142e02918070c49242a18120f6e

File: parser/src/main/parsing/statements/Stmt.c (hunk 1 above)
Comment (pointbazaar) [reply parser/src/main/parsing/statements/Stmt.c:49]: 2c1d6bda81685e0989c50766a5f905c4da254e7c

File: transpiler/main/c_code_gen.c (hunk 2 above)
Comment (pointbazaar) [reply transpiler/main/c_code_gen.c:238]: (not reall) fixed in 3ae5ceebf663834cc74fb46be03e07e5f536da85

File: parser/src/main/commandline/TokenKeys.h [hunk 7]
```diff
@@ -43,5 +43,6 @@
 #define LOOP 36
 #define FN 37
 #define ELSE 38
+#define BREAK 39
```
[the remark this answers is not in the document]
Comment (pointbazaar) [reply parser/src/main/commandline/TokenKeys.h:49]: fixed in e0edcab1b89c435488b1e661d57bb4264fe80124 :+1:

**Review by Milo-D — APPROVED** [review]
PR approved.

**Pull request merged**